pkgs/by-name/tr/treefmt/modules/default.nix

```diff
@@ -0,0 +1,100 @@
+{
+  lib,
+  pkgs,
+  config,
+  options,
+  modulesPath,
+  ...
+}:
+let
+  settingsFormat = pkgs.formats.toml { };
+in
+{
+  _class = "treefmtConfig";
+
+  options = {
+    name = lib.mkOption {
+      type = lib.types.str;
+      default = lib.getName config.package + "-with-config";
+      defaultText = lib.literalExpression "\"\${getName package}-with-config\"";
+      description = ''
+        Name to use for the wrapped treefmt package.
+      '';
+    };
+
+    runtimeInputs = lib.mkOption {
+      type = with lib.types; listOf package;
+      default = [ ];
+      description = ''
+        Packages to include on treefmt's PATH.
+      '';
+    };
+
+    settings = lib.mkOption {
+      type = lib.types.submoduleWith {
+        specialArgs = { inherit modulesPath; };
+        modules = [
+          { freeformType = settingsFormat.type; }
+        ];
+      };
+      default = { };
+      description = ''
+        Settings used to build a treefmt config file.
+      '';
+    };
+
+    configFile = lib.mkOption {
+      type = lib.types.path;
+      # Ensure file is copied to the store
+      apply = file: if lib.isDerivation file then file else "${file}";
```

Comment:
Won't this get copied to the store when `configFile` is referenced below?

Reply:
I may have missed something, but having a `"${}"` in the option's apply function feels like good practice.

This means if the option is a) defined by users to a non-store-path and b) read directly (without using the `result`), it will be coerced into a store-path.

This only really affects users who are using `treefmt.evalConfig` manually, instead of `treefmt.withConfig` or `treefmt.buildConfig`.